src/index.ts

```diff
@@ -0,0 +1,220 @@
+import Config from './config';
+import Logger from './logger';
+import Storage from './storage';
+import Navigator from './navigator';
+import { RadarPublishableKeyError } from './errors';
+
+import AddressesAPI from './api/addresses';
+import ConfigAPI from './api/config';
+import ContextAPI from './api/context';
+import ConversionsAPI from './api/conversions';
+import GeocodingAPI from './api/geocoding';
+import RoutingAPI from './api/routing';
+import SearchAPI from './api/search';
+import TrackAPI from './api/track';
+import TripsAPI from './api/trips';
+
+import MapUI from './ui/map';
+import AutocompleteUI from './ui/autocomplete';
+
+import SDK_VERSION from './version';
+
+import type {
+  Location,
+  RadarAutocompleteParams,
+  RadarConversionParams,
+  RadarDistanceParams,
+  RadarForwardGeocodeParams,
+  RadarMatrixParams,
+  RadarMetadata,
+  RadarOptions,
+  RadarReverseGeocodeParams,
+  RadarSearchGeofencesParams,
+  RadarSearchPlacesParams,
+  RadarTrackParams,
+  RadarTripOptions,
+  RadarValidateAddressParams,
+} from './types';
+
+import 'maplibre-gl/dist/maplibre-gl.css';
+import '../styles/radar.css';
+
+
+const isSecretKey = (key: string): boolean => (
+  key.includes('_sk_')
+);
+const isLiveKey = (key: string): boolean => (
+  key.includes('_live_')
+);
+
+class Radar {
+  public static get VERSION() {
+    return SDK_VERSION;
+  }
+
+  // "ui" namespace
+  public static get ui() {
+    return {
+      maplibregl: MapUI.getMapLibre(),
+      map: MapUI.createMap,
+      marker: MapUI.createMarker,
+      autocomplete: AutocompleteUI.createAutocomplete,
+    };
+  }
+
+  public static initialize(publishableKey: string, options: RadarOptions = {}) {
+    if (!publishableKey) {
+      throw new RadarPublishableKeyError('Publishable key required in initialization.');
+    }
+
+    if (isSecretKey(publishableKey)) {
+      throw new RadarPublishableKeyError('Secret keys are not allowed. Please use your Radar publishable key.');
+    }
+
+    // store settings in global config
+    const live = isLiveKey(publishableKey);
+    const logLevel = live ? 'error' : 'info';
+    const debug = !live;
+    const radarOptions = Object.assign(
+      Config.defaultOptions,
+      {
+        publishableKey,
+        live,
+        logLevel,
+        debug,
+      },
+      options,
+    );
+    Config.setup(radarOptions);
+
+    Logger.info(`initialized with ${live ? 'live' : 'test'} publishableKey.`);
+    if (options.debug) {
+      Logger.info(`using options: ${JSON.stringify(options)}`);
+    }
+
+    ConfigAPI.getConfig();
```

Comment:
since `ConfigAPI.getConfig()` is a `Promise`, should we make `Radar.initialize` a `Promise` as well so that we can give the user the option to `await` for the initialization to complete? (noticed this when writing the tests)

something like 
```suggestion
  public static initialize(publishableKey: string, options: RadarOptions = {}): Promise<void> {
    if (!publishableKey) {
      throw new RadarPublishableKeyError('Publishable key required in initialization.');
    }

    if (isSecretKey(publishableKey)) {
      throw new RadarPublishableKeyError('Secret keys are not allowed. Please use your Radar publishable key.');
    }

    // store settings in global config
    const live = isLiveKey(publishableKey);
    const logLevel = live ? 'error' : 'info';
    const debug = !live;
    const radarOptions = Object.assign(
      Config.defaultOptions,
      {
        publishableKey,
        live,
        logLevel,
        debug,
      },
      options,
    );
    Config.setup(radarOptions);

    Logger.info(`initialized with ${live ? 'live' : 'test'} publishableKey.`);
    if (options.debug) {
      Logger.info(`using options: ${JSON.stringify(options)}`);
    }

    return ConfigAPI.getConfig();
```

Reply:
No, this should happen in the background (does not need to block the call)